code/modules/robotics/bot/secbot.dm

```diff
@@ -1156,7 +1088,77 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
 
 		return
 
+//secbot handcuff bar thing
+/datum/action/bar/icon/secbot_cuff
+	duration = 60
+	interrupt_flags = INTERRUPT_MOVE | INTERRUPT_ACT | INTERRUPT_STUNNED | INTERRUPT_ACTION
+	id = "secbot_cuff"
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "handcuff"
+	var/obj/machinery/bot/secbot/master
+
+	New(var/the_bot)
+		src.master = the_bot
+		..()
 
+	onUpdate()
+		..()
+		if (get_dist(master, master.target) <= 1 || !master.target || master.target.hasStatus("handcuffed"))
+			interrupt(INTERRUPT_ALWAYS)
+			master.process()
+			return
+
+	onStart()
+		..()
+		if (get_dist(master, master.target) <= 1 || !master.target || master.target.hasStatus("handcuffed"))
+			interrupt(INTERRUPT_ALWAYS)
+			master.process()
+			return
+
+		playsound(master, "sound/weapons/handcuffs.ogg", 30, 1, -2)
+		master.mode = SECBOT_ARREST
+		master.visible_message("<span class='alert'><B>[master] is trying to put handcuffs on [master.target]!</B></span>")
+
+	onEnd()
+		..()
+
+		if (get_dist(master, master.target) <= 1)
+			if (!master.target || master.target.hasStatus("handcuffed"))
+				return
+
+			var/uncuffable = 0
+			if (ishuman(master.target))
+				var/mob/living/carbon/human/H = master.target
+				if(!H.limbs.l_arm || !H.limbs.r_arm)
+					uncuffable = 1
+
+			if (!isturf(master.target.loc))
+				uncuffable = 1
+
+			if(ishuman(master.target) && !uncuffable)
+				master.target.handcuffs = new /obj/item/handcuffs(master.target)
+				master.target.setStatus("handcuffed", duration = INFINITE_STATUS)
+
+			var/last_target = master.target
+
+			master.mode = SECBOT_IDLE
+			master.target = null
+			master.anchored = 0
+			master.last_found = world.time
+			master.frustration = 0
+
+			if(!uncuffable)
+				playsound(master.loc, pick('sound/voice/bgod.ogg', 'sound/voice/biamthelaw.ogg', 'sound/voice/bsecureday.ogg', 'sound/voice/bradio.ogg', 'sound/voice/binsult.ogg', 'sound/voice/bcreep.ogg'), 50, 0, 0, 1)
+			if (master.report_arrests && !uncuffable)
+				var/bot_location = get_area(master)
+					//////PDA NOTIFY/////
+				var/datum/radio_frequency/transmit_connection = radio_controller.return_frequency("1149")
+				var/datum/signal/pdaSignal = get_free_signal()
+				pdaSignal.data = list("address_1"="00000000", "command"="text_message", "sender_name"="SECURITY-MAILBOT",  "group"=MGD_SECURITY, "sender"="00000000", "message"="Notification: [last_target] detained by [master] in [bot_location].")
```

Comment:
Just a random thought: If this message included the exact coordinates (of the **cuffed person**, not the bot) somewhere it would allow a competent mechanic to build a contraption that automatically teleports caught criminals to the brig and I like the sound of that.

Reply:
Done! And, there's a setting to make them say a bunch of other obnoxious stuff too.